### ISSUE TITLE
bd52 src contains archived version

### ISSUE THREAD

**Issue by jrvjrv** [vasl-developers/vasl#391]
Most of the boards in vasl/boards/src do not contain an archived version. bd52 does. This has been carried through to the released version (vasl-boards-extensions/boards/bd52), unnecessarily increasing the size of this file.

**Comment by jrvjrv** [comment]
My quick read of the boardbuilder program suggests that it uses a blacklist approach (i.e. it excludes files/file types such as .psd or .xcf ) rather than a whitelist approach, i.e including only files/types it *knows* should be included. It might be nice to change to a whitelist approach.

**Comment by zgrose** [comment]
The board builder doesn't commit files into Git. Should be cleared out with @5001da4

**Comment by jrvjrv** [comment]
Understood, but boardbuilder builds the archive as best I understand the process. If cruft creeps in, with a blacklist process it requires someone to notice. With a whitelist process cruft can't creep in. If the whitelist is not sufficient, in most cases someone will notice relatively soon because a large number of boards will be affected.

**Comment by davidsullivan317** [comment]
The whitelist approach won't work because a number of files do not have extensions.

**Comment by jrvjrv** [comment]
I'm not sure I understand. The other files have fixed names. That's better than matching on extensions. The program I have partially completed (mostly as an exercise in learning Java) uses the following matchers:

        legalFiles.add( new ExactMatch( "data" ) );
        legalFiles.add( new ExtensionMatch( ".gif" ) );
        legalFiles.add( new ExtensionMatch( ".png" ) );
        legalFiles.add( new ExactMatch( "BoardMetadata.xml" ) );
        legalFiles.add( new ExactMatch( "LOSData" ) );
        legalFiles.add( new ExactMatch( "colorSSR" ) );
        legalFiles.add( new ExactMatch( "colors" ) );
        legalFiles.add( new ExactMatch( "overlaySSR" ) );
        legalFiles.add( new ExactMatch( "SSRControls" ) );

This is all I have seen. I have seen a couple of other files/directories that I didn't thing belonged (thumbs.db, __MacOS.something, oldData, README*.txt), but I didn't file bugs for them because they didn't add that much to the file size. The list above seems to cover the current list of "correct" files that should be found in map and overlay archives.

**Comment by zgrose** [comment]
I wonder if it would be practical to enhance this idea into a better .gitignore file so that not only are the boards cleaner but the repo as well? That's the root cause of things getting into GitHub, after all. What we package up is a parallel problem.

**Comment by jrvjrv** [comment]
I would suggest having the build process gather what it needs rather than trying to restrict the repository. To me it's nice to have the source graphics (*.xcf, *.psd) right there in the directory, and you might want to have notes (README.txt) or other artifacts in the source directory too. A build process should ignore everything but what it needs to create the board archive. If the person building the map archives has left something around inadvertently, it won't affect the build. Ideally the build process should also call out when things it needs are missing, e.g. an archive missing a "data" file. 

I looked briefly at the boardbuilder program, and I wonder if it might not be better to use a build tool instead. I have learned a bit about maven, but I am not sure it would be easy to adapt it to this problem. To me it looks like gradle might be better for this case.

**Comment by zgrose** [comment]
This very issue you filed was to remove an unwanted file from the src directory (the bdX file inside of the bdX src folder). Since I obviously can't convince you that controlling the contents of Git and controlling the contents of the board file are two separate issues, I'll just wait on your code changes and take it from there.
